tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/BuilderUtil.scala

```diff
@@ -52,4 +52,32 @@ object BuilderUtil {
 
   /** generates a BuilderTypeException wrapped in a Left, containing the given message */
   def throwMsg(msg: String): TypeComputationResult = Left(new TBuilderTypeException(msg))
+
+  /** Generates a (total) signature from a partial one, by returning a Left outside the domain */
+  def completePartial(name: String, partialSig: PartialSignature): Signature = {
```

Comment:
I'd hold off on designating `private/protected` until all the sub-builders are written, since a few of the `TypeComputation`s, especially for function/record operators are going to be less formulaic and will probably use fewer composition methods like `mkSigMapEntry` and more low-level ones.